pkgs/by-name/me/meson/package.nix

```diff
@@ -88,8 +93,16 @@ python3.pkgs.buildPythonApplication rec {
     LDAP
     OpenGL
     openldap
+  ] ++ lib.optionals (stdenv.cc.isClang && !stdenv.isDarwin) [
+    libblocksruntime
+    llvmPackages.openmp
   ];
 
+  postPatch = lib.optionalString stdenv.hostPlatform.isFreeBSD ''
+    # This edge case is explicilty part of meson but is wrong for nix
+    sed -E -i -e s/mesonlib.is_freebsd../False/g mesonbuild/modules/pkgconfig.py
```

Comment:
Fine to patch it this way, if it's a failure also visible at build time.

Reply:
The failure manifests at build time not for meson but for packages using meson. Is that what you meant?

Reply:
Ok. Ideally we would be not able to build the package at all if the code changes but the error cannot be captured.

Mind using this patch instead? Afaik this can be even used unconditionally.

```
diff --git a/mesonbuild/modules/pkgconfig.py b/mesonbuild/modules/pkgconfig.py
index ebe0d92d5..99f84f8e3 100644
--- a/mesonbuild/modules/pkgconfig.py
+++ b/mesonbuild/modules/pkgconfig.py
@@ -693,7 +693,7 @@ def parse_variable_list(vardict: T.Dict[str, str]) -> T.List[T.Tuple[str, str]]:
         pcfile = filebase + '.pc'
         pkgroot = pkgroot_name = kwargs['install_dir'] or default_install_dir
         if pkgroot is None:
-            if mesonlib.is_freebsd():
+            if False:
                 pkgroot = os.path.join(_as_str(state.environment.coredata.get_option(mesonlib.OptionKey('prefix'))), 'libdata', 'pkgconfig')
                 pkgroot_name = os.path.join('{prefix}', 'libdata', 'pkgconfig')
             elif mesonlib.is_haiku():

```

Reply:
Ohhhhh, I understand what you're saying now.

That's fine, sure. I agree with the aim of using a diff file, and I only made it conditional to avoid mass rebuilds on the linux side.

Reply:
It also needs to be conditional, is what I am realizing now.